src/rust/src/x509/verify.rs

```diff
@@ -43,6 +47,125 @@ impl CryptoOps for PyCryptoOps {
     }
 }
 
+struct FixedPolicy<'a>(Policy<'a, PyCryptoOps>);
+
+/// This enum exists solely to provide heterogeneously typed ownership for `OwnedPolicy`.
+enum SubjectOwner {
+    // TODO: Switch this to `Py<PyString>` once Pyo3's `to_str()` preserves a
+    // lifetime relationship between an a `PyString` and its borrowed `&str`
+    // reference in all limited API builds. PyO3 can't currently do that in
+    // older limited API builds because it needs `PyUnicode_AsUTF8AndSize` to do
+    // so, which was only stabilized with 3.10.
+    DNSName((pyo3::Py<pyo3::PyAny>, String)),
+    IPAddress((pyo3::Py<pyo3::PyAny>, pyo3::Py<pyo3::types::PyBytes>)),
```

Comment:
I think it'd be marginally cleaner if there was a seperate `py_subject` field on the Verifier, and then `SubjectOwner`'s enum elements only needed to be a single field.

Reply:
https://github.com/pyca/cryptography/pull/9405/commits/3c036bcd157c0ccb92d50e1a8e44d6099e970fe7